src/cmd/plumb/rules.c

```diff
@@ -244,16 +244,37 @@ filename(Exec *e, char *name)
 	return cleanname(buf);
 }
 
+static char*
+subexpmatch(Exec *e, char *s, int *numlen)
+{
+	int n, d, ok;
+	char *t;
+
+	n = 0;
+	ok = 1;
+	for(t = s; '0'<=*t && *t<='9'; t++)
```

Comment:
This feels an awful lot like `strtol`.

Something like,

    t = nil;
    long n = strtol(s, &t, 10);
    *numlen = t-s;
    if(t==s||0>n||n>NMATCHSUBEXP)
        return nil;
    return e->match[n];

? (untested, modulo silly errors)

Reply:
It does—just for the case of unsigned string-of-digits-only numerals.
Since this special case is simple, and the resulting inline parsing
turned out short enough, I decided to leave it. (Plus, there are
already a few places in the tree, where this very special case is
also handled inline, so it did not feel that out of place.)

Switching to `strtol` is fine with me.

I’ve updated the pull request to call to `strotoul` instead. It’s `strotoul`
and not `strtol` to avoid checking for a negative return value: even
though we cannot have a negative value here (as we know it starts
with a digit), missing the check would stick out as a potential mistake
when skimming the code. For the same reason, I still check for `t==s`.